plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java

```diff
@@ -636,62 +656,283 @@ public Stream<SourceFile> parse(Project subproject, Set<Path> alreadyParsed, Exe
+                            .collect(toList());
+
+                    alreadyParsed.addAll(groovyPaths);
+
+                    Stream<SourceFile> cus = Stream.of((Supplier<GroovyParser>) () -> GroovyParser.builder()
+                            .classpath(dependenciesWithBuildDirs)
+                            .styles(styles)
+                            .typeCache(javaTypeCache)
+                            .logCompilationWarningsAndErrors(false)
+                            .build()).map(Supplier::get).flatMap(gp -> gp.parse(groovyPaths, baseDir, ctx)).map(cu -> {
+                        if (isExcluded(exclusions, cu.getSourcePath()) || cu.getSourcePath().startsWith(buildDir)) {
+                            return null;
+                        }
+                        return cu;
+                    }).filter(Objects::nonNull).map(it -> it.withMarkers(it.getMarkers().add(javaVersion)));
+                    sourceSetSourceFiles = Stream.concat(sourceSetSourceFiles, cus);
+                    sourceSetSize += groovyPaths.size();
+                    logger.info(
+                            "Scanned {} Groovy sources in {}/{}",
+                            groovyPaths.size(),
+                            subproject.getPath(),
+                            sourceSet.getName());
+                }
+            }
+
+            for (File resourcesDir : sourceSet.getResources().getSourceDirectories()) {
+                if (resourcesDir.exists() && !alreadyParsed.contains(resourcesDir.toPath())) {
+                    OmniParser omniParser = omniParser(alreadyParsed, subproject);
+                    List<Path> accepted = omniParser.acceptedPaths(baseDir, resourcesDir.toPath());
+                    sourceSetSourceFiles = Stream.concat(
+                            sourceSetSourceFiles,
+                            omniParser.parse(accepted, baseDir, new InMemoryExecutionContext())
+                                    .map(it -> it.withMarkers(it.getMarkers().add(javaVersion))));
+                    alreadyParsed.addAll(accepted);
+                    sourceSetSize += accepted.size();
+                }
+            }
+
+            JavaSourceSet sourceSetProvenance = JavaSourceSet.build(sourceSet.getName(), dependencyPaths);
+            sourceFileStream = sourceFileStream.concat(
+                    sourceSetSourceFiles.map(addProvenance(sourceSetProvenance)),
+                    sourceSetSize);
+            // Some source sets get misconfigured to have the same directories as other source sets
+            // Prevent files which appear in multiple source sets from being parsed more than once
+            for (File file : sourceSet.getAllSource().getSourceDirectories().getFiles()) {
+                alreadyParsed.add(file.toPath());
+            }
+        }
+        return sourceFileStream;
+    }
+
+    private SourceFileStream parseAndroidProjectSourceSets(Project subproject,
```

Comment:
callout: Future problem will almost certainly be needing an Android specific marker to carry various information such as the min and max SDK versions, etc.